libethereum/BlockChain.cpp

```diff
@@ -206,18 +206,15 @@ bool BlockChain::open(fs::path const& _path, WithExisting _we)
     {
         if (_we == WithExisting::Kill)
         {
-            cnote << "Killing blockchain (" << chainSubPathBlocks << ") & extras ("
-                  << extrasSubPathExtras << ") databases (WithExisting::Kill).";
-            fs::remove_all(chainSubPathBlocks);
-            fs::remove_all(extrasSubPathExtras);
+            LOG(m_loggerInfo) << "Deleting all databases. This will require a resync from genesis.";
+            fs::remove_all(chainPath);
```

Comment:
I think here it was removing only blocks and extras because `BlockChain` class is responsible on only for these DB, and `State` is responsible for state.

I'd leave it without change for better separation of concertns.
(with this change there's at least a problem when it gets to `State::openDB` and logs `Killing state database` - that's not true anymore)

Reply:
Ah I see below that you did it to remove also extras.old, if it's there... Maybe better to remove existing extras.old in that case?

Reply:
I think we should leave extras.old (unless one explicitly specifies --kill) since there's the possibility of using it to restore your extras DB (if you know what you're doing), keeping it around gives the user the chance to do that.